src/gamepad/mapping.vala

```diff
@@ -0,0 +1,196 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class LibGamepad.Dpad : Object {
+	public InputType types[4];
+	public int values[4];
+	public int axisval[2];
+
+	public Dpad () {
```

Comment:
as we don't do anything special maybe it's better to make it a construct block (just replace `public Dpad ()` by `construct`).